OpenRA.Game/Network/Connection.cs

```diff
@@ -243,4 +245,40 @@ protected override void Dispose(bool disposing)
 			base.Dispose(disposing);
 		}
 	}
+
+	public class ConnectionAddress
+	{
+		public readonly string Host;
+		public readonly int Port;
+		public readonly bool IsValid;
+		public readonly bool IsIpV6;
+
+		public ConnectionAddress(string address) : this(address, 0)
+		{
+		}
+
+		public ConnectionAddress(string host, int port)
+		{
+			if (port == 0 && new Regex(":[0-9]+$").IsMatch(host))
+			{
+				var portPosition = host.LastIndexOf(":", StringComparison.Ordinal);
+				int.TryParse(host.Substring(portPosition + 1), out port);
+				host = host.Substring(0, portPosition);
+			}
+
+			if (host.StartsWith("[") && host.EndsWith("]"))
+				host = host.Substring(1, host.Length - 2);
+
+			var type = Uri.CheckHostName(host);
+			IsIpV6 = type == UriHostNameType.IPv6;
+			Host = host;
+			Port = port;
+			IsValid = type != UriHostNameType.Unknown && Port > 0x0000 && Port < 0xffff;
```

Comment:
Port 0 is valid but reserved by IANA. Port 65535 is valid and allocated for dynamic / private ports. While we can discuss about port 0, port 65535 has to be accepted.

Reply:
ports are defined with UInt16.